cstate/powermgr_cstate_tests.sh

```diff
@@ -0,0 +1,322 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-2.0-only
+# Copyright (c) 2022 Intel Corporation
+# @Author   wendy.wang@intel.com
+# @Desc     Test script to verify Intel CPU Core Cstate functionality
+# @History  Created Nov 01 2022 - Created
+
+cd "$(dirname $0)" 2>/dev/null
+source ../.env
+source common.sh
+
+CPU_SYSFS_PATH="/sys/devices/system/cpu"
+CPU_BUS_SYSFS_PATH="/sys/bus/cpu/devices/"
+CPU_IDLE_SYSFS_PATH="/sys/devices/system/cpu/cpuidle"
+
+current_cpuidle_driver=$(cat "$CPU_IDLE_SYSFS_PATH"/current_driver)
+
+#Turbostat tool is required to run core cstate cases
+turbostat sleep 1 1>/dev/null 2>&1 || skip_test "Turbostat tool is required to \
+to run CPU core cstate cases, please get it from latest upstream kernel-tools."
+
+#Funtion to verify if Intel_idle driver refer to BIOS _CST table
+test_cstate_table_name() {
+    local cstate_name
+    local name
+
+    cstate_name=$(cat "$CPU_SYSFS_PATH"/cpu0/cpuidle/state*/name)
+    name=$(echo "$cstate_name" | grep ACPI)
+    if [[ -n $name ]]; then
+        test_print_trc "$cstate_name"
+        die "Intel_idle driver refers to ACPI cstate table."
+    else
+        test_print_trc "$cstate_name"
+        test_print_trc "Intel_idle driver refers to BIOS _CST table."
+    fi
+}
+
+#Funtion to verify if current idle driver is intel_idle
+check_intel_idle() {
+    [[ $current_cpuidle_driver == "intel_idle" ]] || {
+        block_test "If the platform does not support Intel_Idle driver yet, \
+please ignore this test case."
+    }
+}
+
+#Function to switch each core cstate
+test_cstate_switch_idle() {
+    local usage_before=()
+    local usage_after=()
+    local CPUS
+    CPUS=$(ls "$CPU_BUS_SYSFS_PATH" | xargs)
+    local cpu_num
+    cpu_num=$(lscpu | grep "^CPU(s)" | awk '{print $2}')
+
+    if [[ -n "$CPUS" ]]; then
+        for cpu in $CPUS; do
+            STATES=$(ls "${CPU_BUS_SYSFS_PATH}"/"${cpu}"/cpuidle | grep state | xargs)
```

Comment:
STATES  should be states with above reason.

Reply:
Resolved, thanks!